public/app/features/datasources/components/EditDatasSourceSubtitle.tsx

```diff
@@ -0,0 +1,81 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Badge, InlineField, InlineSwitch, useStyles2 } from '@grafana/ui';
+
+interface Props {
+  subTitle?: string;
+  isDefault: boolean;
+  alertingSupported: boolean;
+  onDefaultChange: (value: boolean) => void;
+}
+
+export function EditDataSourceSubtitle({ subTitle, isDefault, alertingSupported, onDefaultChange }: Props) {
+  const styles = useStyles2(getStyles);
+  const dataSourceType = subTitle ? getDataSourceType(subTitle) : '';
+  return (
+    <div className={styles.subTitleRow}>
+      <div className={styles.subTitleRowGroupFirst}>
+        <div className={styles.subTitle}>
+          Type: <span>{dataSourceType}</span>
+        </div>
+        <InlineField
+          label="Default"
+          tooltip="This datasource is used when you select the data source in panels. The default data source is
+        'preselected in new panels."
+          transparent={true}
+          labelWidth={8}
+          disabled={false}
+          className={styles.defaultDataSourceSwitch}
+        >
+          <InlineSwitch
+            id="basic-settings-default"
+            transparent={true}
+            value={isDefault}
+            onChange={(event: React.FormEvent<HTMLInputElement>) => onDefaultChange(event.currentTarget.checked)}
+          />
+        </InlineField>
+      </div>
+      {alertingSupported ? (
+        <Badge color="green" icon="check-circle" text="Alerting supported" />
+      ) : (
+        <Badge color="orange" icon="exclamation-triangle" text="Alerting not supported" />
+      )}
+    </div>
+  );
+}
+
+const getDataSourceType = (subTitle: string) => {
+  const typeArray = subTitle.split('Type: ');
+  return typeArray[1];
```

Comment:
This makes this code very fragile to how we format the subtitle in useDataSourceSettingsNav. please pass the plugin name instead via props .